paddle/fluid/operators/set_value_op.h

```diff
@@ -292,5 +312,255 @@ class SetValueKernel : public framework::OpKernel<T> {
+                        decrease_axis, starts.size());
+
+    for (size_t axis = 0; axis < D; axis++) {
+      starts_indices[axis] = 0;
+      ends_indices[axis] = out_dims[axis];
+      steps_indices[axis] = 1;
+      reverse_axis[axis] = false;
+    }
+
+    for (size_t axis = 0; axis < axes.size(); axis++) {
+      int axis_index = axes[axis];
+      starts_indices[axis_index] = starts[axis];
+      ends_indices[axis_index] = ends[axis];
+      steps_indices[axis_index] = steps[axis];
+      reverse_axis[axis_index] = (reverse_vector[axis] == 1) ? true : false;
+    }
+
+    bool need_reverse = false;
+    for (size_t axis = 0; axis < axes.size(); axis++) {
+      if (reverse_vector[axis] == 1) {
+        need_reverse = true;
+        break;
+      }
+    }
+
+    auto& dev_ctx = context.template device_context<DeviceContext>();
+    auto& place =
+        *context.template device_context<DeviceContext>().eigen_device();
+    math::SetConstant<DeviceContext, T> set_zero;
+
+    if ((grad_value) && (!grad_value->IsInitialized())) {
+      grad_value->mutable_data<T>(context.GetPlace());
+    }
+    if (grad_input) {
+      // Set gradient of `Input`
+      TensorCopy(*in, context.GetPlace(), grad_input);
+
+      auto grad_input_t =
+          framework::EigenTensor<T, D, Eigen::RowMajor,
+                                 Eigen::DenseIndex>::From(*grad_input);
+
+      framework::Tensor tmp(grad_input->type());
+      tmp.mutable_data<T>(out_dims, context.GetPlace());
+      set_zero(dev_ctx, &tmp, static_cast<T>(0));
+      auto tmp_t = framework::EigenTensor<T, D, Eigen::RowMajor,
+                                          Eigen::DenseIndex>::From(tmp);
+
+      grad_input_t.stridedSlice(starts_indices, ends_indices, steps_indices)
+          .device(place) = tmp_t;
+    }
+    if (grad_value) {
+      set_zero(dev_ctx, grad_value, static_cast<T>(0));
+
+      auto in_t = framework::EigenTensor<T, D, Eigen::RowMajor,
+                                         Eigen::DenseIndex>::From(*in);
+
+      if (grad_value->dims() == out_dims) {
+        auto grad_value_t =
+            framework::EigenTensor<T, D, Eigen::RowMajor,
+                                   Eigen::DenseIndex>::From(*grad_value);
+        if (need_reverse) {
+          framework::Tensor tmp(grad_value->type());
+          tmp.mutable_data<T>(out_dims, context.GetPlace());
+          set_zero(dev_ctx, &tmp, static_cast<T>(0));
+          auto tmp_t = framework::EigenTensor<T, D, Eigen::RowMajor,
+                                              Eigen::DenseIndex>::From(tmp);
+
+          tmp_t.device(place) =
+              in_t.stridedSlice(starts_indices, ends_indices, steps_indices);
+          grad_value_t.device(place) = tmp_t.reverse(reverse_axis);
+        } else {
+          grad_value_t.device(place) =
+              in_t.stridedSlice(starts_indices, ends_indices, steps_indices);
+        }
+      } else {
```

Comment:
what case grad_value->dims() != out_dims is ?

Reply:
eg: `t[0] = value`, `t[0].shape` is `[3,3]`, `value.shape` is `[1,]`,